### ISSUE TITLE
make AntiforgeryMiddleware extendable

### ISSUE THREAD

**Issue by yousefak007** [FastEndpoints/FastEndpoints#776]
hi, could you please make AntiforgeryMiddleware extendable?

I want to add a check for Origin to disable CSRF 
because on API or Mobile, we don't need to use Antiforgery

**Comment by yousefak007** [comment]
I found this solution. 

```c#
app.UseWhen(context => !context.Request.Headers.Origin.IsNullOrEmpty(), app => app.UseAntiforgeryFE())
```

But I think it's better to expose the FE Api to facilitate the modification or extend it

**Comment by dj-nitehawk** [comment]
implemented in `v5.29.0.12-beta`

```cs
.UseAntiforgeryFE(httpCtx => !httpCtx.Request.Headers.Origin.IsNullOrEmpty())
```
